src/bin/nydus-image/inspect.rs

```diff
@@ -141,20 +181,19 @@ impl RafsInspector {
 
     fn load_meta(f: &mut RafsIoReader) -> Result<(RafsMeta, RafsLayout)> {
         let layout_profile = RafsLayout::rafsv5_layout();
-        match Self::super_block_v5(f, &layout_profile) {
-            Ok(sb) => {
+        let mut res = Self::super_block_v5(f, &layout_profile).map(|sb| {
```

Comment:
`super_block_v5` does `RafsV5SuperBlock::load()` which doesn't check fs magic or fs version, so it won't return errors if it's a v6 superblock.

Reply:
I modified the super_block_v5 and added a checker for rafs versions.

```rust
      if !sb.detect() {
            return Err(anyhow!("invalid rafs v5 version"));
        }
```